### PR TITLE
Use eslint with loopback config

### PR DESCRIPTION
**Pull request by gunjpan**
- Introduce eslint infrastructure and fix linting errors
- Remove jshint & jscrc references
- Update dependencies in package.json

Connect to strongloop/loopback#1749

### CODE REVIEW

**Comment by gunjpan** [comment]
@bajtos: PTAL at the commits, thank you.

File: example/documentation/remotes/simple.js [hunk 1]
```diff
@@ -17,7 +17,10 @@ function getSecret(callback) {
  * Takes a string and returns an updated string.
  */
 helper.method(transform, {
-  accepts: [{ name: 'str', type: 'string', required: true, description: 'The value to update' }],
+  accepts: [{ name: 'str',
+              type: 'string',
+              required: true,
+              description: 'The value to update' }],
```
Comment (bajtos) [inline example/documentation/remotes/simple.js]: IIRC, we are using a different formating in loopback - put just the `description` on the new line:

```js
{
  accepts: [{ name: 'str', type: 'string', required: true,
    description: 'The value to update' }],
}
```

File: test/rest.test.js [hunk 2]
```diff
@@ -1592,22 +1586,23 @@ describe('strong-remoting-rest', function() {
           .end(expectErrorResponseContaining({ message: 'an error' }, done));
       });
 
-      it('should return 500 for unhandled errors thrown from before hooks', function(done) {
-        var method = givenSharedStaticMethod();
+      it('should return 500 for unhandled errors thrown from before hooks',
+        function(done) {
```
Comment (bajtos) [inline test/rest.test.js]: To keep amount of white-space changes smaller, please use the following formatting:

```js
it('should return 500 for unhandled errors thrown from before hooks',
function(done) {
  // original code - no indentation change needed
});
```

**Comment by bajtos** [comment]
@gunjpan looks mostly good, see few comments above.

@superkhau would you like to take a look too?

File: package.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
     "tag": "next"
   },
   "scripts": {
-    "pretest": "grunt jscs jshint",
+    "pretest": "grunt eslint",
```
Comment (superkhau) [inline package.json]: posttest

**Comment by superkhau** [comment]
Please change `pretest` to posttest`, otherwise LGTM.

**Comment by gunjpan** [comment]
@bajtos @superkhau : Updated, PTAL. Will rebase & merge when LGTM'ed.

**Comment by bajtos** [comment]
@slnode test please

**Pull request merged**